extensions/qute/runtime/src/main/java/io/quarkus/qute/runtime/ContentTypes.java

```diff
@@ -0,0 +1,50 @@
+package io.quarkus.qute.runtime;
+
+import java.net.URLConnection;
+
+import javax.inject.Inject;
+import javax.inject.Singleton;
+
+import org.jboss.logging.Logger;
+
+import io.quarkus.qute.Variant;
+
+@Singleton
+public class ContentTypes {
+
+    private static final Logger LOGGER = Logger.getLogger(ContentTypes.class);
+
+    @Inject
+    QuteConfig config;
+
+    /**
+     * 
+     * @param templatePath The path relative to the template root, uses the {@code /} path separator.
+     * @return the content type
+     */
+    public String getContentType(String templatePath) {
+        String fileName = templatePath;
+        int slashIdx = fileName.lastIndexOf('/');
+        if (slashIdx != -1) {
+            fileName = fileName.substring(slashIdx, fileName.length());
+        }
+        int dotIdx = fileName.lastIndexOf('.');
+        if (dotIdx != -1) {
+            String suffix = fileName.substring(dotIdx + 1, fileName.length());
+            String additionalContentType = config.contentTypes.get(suffix);
+            if (additionalContentType != null) {
+                return additionalContentType;
+            }
+            if (suffix.equalsIgnoreCase("json")) {
+                return Variant.APPLICATION_JSON;
+            }
+            String contentType = URLConnection.getFileNameMap().getContentTypeFor(fileName);
```

Comment:
We could use `Files.probeContentType` too (or instead?) to take advantage of libraries that register their own `java.nio.file.spi.FileTypeDetector` through the ServiceLoader mechanism

Reply:
I was thinking about `Files.probeContentType()` too but given the fact that the result may differ, based on the OS or presence of other libraries, I decided to choose the simple and straightforward solution for now. Also it seems that the default `FileTypeDetector` impl is makes use of `URLConnection.getFileNameMap()` anyway...